pkg/endpoint/regenerator.go

```diff
@@ -45,34 +55,66 @@ func newRegenerator(in struct {
 
 	Logger *slog.Logger
 
-	Config      wait.TimeoutConfig
-	NodesWaitFn KVStoreNodesWaitFn
-	IPCacheSync *ipcache.IPIdentityWatcher
-	ClusterMesh *clustermesh.ClusterMesh
+	Config         wait.TimeoutConfig
+	NodesWaitFn    KVStoreNodesWaitFn
+	IPCacheSync    *ipcache.IPIdentityWatcher
+	ClusterMesh    *clustermesh.ClusterMesh
+	Fence          regeneration.Fence
+	LBInitWaitFunc loadbalancer.InitWaitFunc
 }) *Regenerator {
 	waitFn := func(context.Context) error { return nil }
 	if in.ClusterMesh != nil {
 		waitFn = in.ClusterMesh.IPIdentitiesSynced
 	}
-
-	return &Regenerator{
+	r := &Regenerator{
 		logger:        in.Logger,
 		nodesWaitFn:   in.NodesWaitFn,
 		ipcacheWaitFn: in.IPCacheSync.WaitForSync,
 		cmWaitFn:      waitFn,
 		cmWaitTimeout: in.Config.ClusterMeshSyncTimeout,
+		fence:         in.Fence,
 	}
+
+	// Wait for ipcache sync before regeneration for endpoints including
```

Comment:
I understand that this PR is intended to be somewhat minimal, so this makes sense. I would eventually like to see this moved to the individual kvstore / clustermesh cells. Perhaps a `//TODO` to that effect, or a comment not to add any more waits here?

Reply:
Fair enough. Added a comment.

Reply:
monkey see, monkey do :-)